### PR TITLE
Grab Current Tracing Fake Mode in a couple spots

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99377

Fix for https://github.com/pytorch/pytorch/issues/99286. There were a couple locations we were instantiating new fake modes instead of grabbing the correct one from the current tracing context/inputs.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/fx_passes/joint_graph.py [hunk 1]
```diff
@@ -14,7 +15,7 @@
 def lazy_init():
     from .fuse_attention import _sfdp_init
 
-    with FakeTensorMode():
+    with detect_fake_mode() or FakeTensorMode():
```
Comment (ezyang) [inline torch/_inductor/fx_passes/joint_graph.py:18]: This is wrong as the patterns here are shared globally so you shouldn't arbitrarily pick the fake mode from a particular run; I actually have the more correct fix on a branch I am working on

Comment (eellison) [reply torch/_inductor/fx_passes/joint_graph.py:18]: Okay - I'm going to close and re-assign to you then.

Comment (vkuzo) [reply torch/_inductor/fx_passes/joint_graph.py:18]: @ezyang  could you share which PR? I want to follow when it lands as I have this PR checked out locally to unblock some quantization experiments.

**Pull request closed without merging**